model/host/db.go

```diff
@@ -464,3 +465,21 @@ func statsByDistroPipeline() []bson.M {
 		},
 	}
 }
+
+// QueryWithFullTaskPipeline returns a pipeline to match hosts and embeds the
+// task document within the host, if it's running a task
+func QueryWithFullTaskPipeline(match bson.M) []bson.M {
+	return []bson.M{
+		{
+			"$match": match,
+		},
+		{
+			"$lookup": bson.M{
+				"from":         task.Collection,
+				"localField":   RunningTaskKey,
+				"foreignField": task.IdKey,
+				"as":           "task_full",
+			},
+		},
```

Comment:
Should there also be a project stage to reduce the amount of data returned? What fields will we actually use?

Reply:
We use 6 fields from the task document in the UI. If we want to remove unneeded fields that we send to the UI we can do it there, but it won't make a noticeable improvement